pkg/lib/scopedclient/warning.go

```diff
@@ -0,0 +1,53 @@
+package scopedclient
+
+import (
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/rest"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/metrics"
+)
+
+// WarningRoundTripper implements the http RouterTripper interface and alerts via metrics if warning
+// headers are present in the response, such as deprecation warnings (299). If the alert (via a metric) fails no error is returned
+// as the RoundTrip interface should not return an error unless a transport level issue occurs.
+type WarningRoundTripper struct{}
+
+func SetWarningRoundTripper(config *rest.Config) *rest.Config {
+	if config != nil {
+		config.Transport = new(WarningRoundTripper)
+	}
+	return config
+}
+
+func (w *WarningRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
```

Comment:
Ah, I was hoping we could do that. But it makes sense, since this is just part of the path of the response, we should not read or close it directly (we could copy the data and do whatever on that though). 

I can checkout the idea of reversing the URL to get the resource info (not to unmarshal the response directly). 

Putting code into the stepper would work too, but feels like quite a bit of complexity